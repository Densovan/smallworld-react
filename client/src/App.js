import React, { Component } from "react"
import { BrowserRouter as Router, Switch, Redirect, Route } from "react-router-dom"
import Index from "./components/index"
import AboutPage from "./components/about"
import WhatWeCanDo from "./components/what-we-can-do"
import Contact from "./components/contact"
import News from "./components/news-and-events"

class App extends Component {
  render() {
    return (
      <Router>
        <Switch>
          <Route exact path="/" component={Index} />
          <Route exact path="/about-us" component={AboutPage} />
          <Route exact path="/what-we-do" component={WhatWeCanDo} />
          <Route exact path="/news-and-events" component={News} />
          <Route exact path="/contact-us" component={Contact} />
          <Redirect to="/" component={Index} />
        </Switch>
      </Router>
    )
  }
}

export default App
