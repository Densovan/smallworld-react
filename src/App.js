import React, { Component } from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import Index from "./components/index";
import AboutPage from "./components/about";
import WhatWeCanDo from "./components/what-we-can-do";
import Contact from "./components/contact";
import News from "./components/news-and-events";

class App extends Component {
  render() {
    window.scrollTo(0, 0);
    return (
      <div className="App">
        <Switch>
          <Route exact path="/" component={Index} />
          <Route exact path="/about-us" component={AboutPage} />
          <Route exact path="/what-we-do" component={WhatWeCanDo} />
          <Route exact path="/news-and-events" component={News} />
          <Route exact path="/contact-us" component={Contact} />
          <Redirect to="/" component={Index} />
        </Switch>
      </div>
    );
  }
}


export default App;
