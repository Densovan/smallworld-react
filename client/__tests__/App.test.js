import React from "react"
import { shallow } from "enzyme"
import App from "../src/App"
import toJson from 'enzyme-to-json'; //added this line


describe("App component", () => {
  
  it("App renders without crashing", () => {
    const wrapper = shallow(<App/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
})
