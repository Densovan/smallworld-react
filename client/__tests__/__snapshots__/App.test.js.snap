// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App renders without crashing 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/about-us"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/what-we-do"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/news-and-events"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/contact-us"
    />
    <Redirect
      component={[Function]}
      to="/"
    />
  </Switch>
</BrowserRouter>
`;
